File: matlive-sdk/src/main/java/com/matnsolutions/matlive_sdk/audio/define/mat_live_user.dart
import 'package:flutter/material.dart';

class MatLiveUser {
  MatLiveUser({
    required this.userId,
    required this.name,
    required this.avatar,
    required this.roomId,
    this.metadata,
  });

  late String userId;
  late String name;
  late String avatar;
  late String roomId;
  String? metadata;

  String? streamID;
  int viewID = -1;
  ValueNotifier<Widget?> videoViewNotifier = ValueNotifier(null);
  ValueNotifier<bool> isCamerOnNotifier = ValueNotifier(false);
  ValueNotifier<bool> isMicOnNotifier = ValueNotifier(true);
  late ValueNotifier<String?> avatarUrlNotifier = ValueNotifier(avatar);
}

enum MatLiveUserRole {
  audience,
  coHost,
  host,
}

File: matlive-sdk/src/main/java/com/matnsolutions/matlive_sdk/audio/define/mat_live_user.kt
package com.matnsolutions.matlive_sdk.audio.define

import androidx.lifecycle.MutableLiveData

data class MatLiveUser(
    var userId: String,
    var name: String,
    val avatar: String,
    var roomId: String,
    val metadata: String? = null
) {
    var streamID: String? = null
    var viewID: Int = -1
    var videoViewNotifier: MutableLiveData<Any?> = MutableLiveData(null)
    var isCameraOnNotifier: MutableLiveData<Boolean> = MutableLiveData(false)
    var isMicOnNotifier: MutableLiveData<Boolean> = MutableLiveData(true)
    var avatarUrlNotifier: MutableLiveData<String?> = MutableLiveData(avatar)
}

enum class MatLiveUserRole {
    audience,
    coHost,
    host
}